frontend/src/metabase/hooks/use-is-truncated.ts

```diff
@@ -0,0 +1,36 @@
+import { useLayoutEffect, useRef, useState } from "react";
+
+import resizeObserver from "metabase/lib/resize-observer";
+
+export const useIsTruncated = <E extends HTMLElement>() => {
```

Comment:
Can it be `Element` and not `HTMLElement`?

Reply:
If we make it `Element`, then we cannot use `ofsetWidth`, we'd have to go with `clientWidth`. 

There is an inconsistency in `getIsTruncated` (I didn't change it in this PR):
- for height we're checking `clientHeight`
- for width we're checking `offsetWidth` (not `clientWidth`)

I think this may have been an intentional decision (to account for the scrollbar) but I found no info about it in commit history.

Nonetheless, I think we should be using `clientWidth`. This change will affect all places `Ellipsified` is used in, though. 
And then we can use `Element` here. 

Good to go?

Reply:
72e61b5c99b5315c104940ddf434cc9b2f409701